arithmetization/src/main/java/net/consensys/linea/zktracer/types/MemoryRange.java

```diff
@@ -73,21 +73,21 @@ public MemoryRange(long contextNumber) {
   public MemoryRange(final long contextNumber, final Range range, final Bytes rawData) {
     this.contextNumber = contextNumber;
     this.range = range;
-    this.rawData = rawData;
+    this.rawData = range.isEmpty() ? Bytes.EMPTY : rawData;
   }
 
   public MemoryRange(
       final long contextNumber, final long offset, final long size, final Bytes rawData) {
     this.contextNumber = contextNumber;
     this.range = Range.fromOffsetAndSize(offset, size);
-    this.rawData = rawData;
+    this.rawData = range.isEmpty() ? Bytes.EMPTY : rawData;
```

Comment:
yes

Reply:
Although this is a constructor ... I'm not sure if you can "already" call the `isEmpty()` method here 🤔